test/csp.js

```diff
@@ -37,8 +37,61 @@ describe('CSP', function () {
 
     request(app.listen())
     .get('/')
-    .expect('Content-Security-Policy', 'default-src *; ')
+    .expect('Content-Security-Policy', 'default-src *')
     .expect('hello')
     .expect(200, done);
   });
+
+  it('string config', function (done) {
+    var app = mock({
+      csp: {
+        policy: 'default-src *'
+      }
+    });
+
+    app.get('/', function (req, res) {
+      res.status(200).end();
+    });
+
+    request(app.listen())
+    .get('/')
+    .expect('Content-Security-Policy', 'default-src *')
+    .expect(200, done);
+  });
+
+  it('array config', function (done) {
+    var app = mock({
+      csp: {
+        policy: ['default-src *', 'img-src *']
+      }
+    });
+
+    app.get('/', function* () {
+      this.body = 'hello';
+    });
+
+    request(app.listen())
+    .get('/')
+    .expect('Content-Security-Policy', 'default-src *; img-src *')
+    .expect(200, done);
+  });
+
+  it('nested config', function (done) {
+    var app = mock({
+      csp: {
+        policy: [
+        { 'default-src': '*' },
+        'img-src *'
+        ]
+      }
+    });
```

Comment:
I'm using mocks directly in test since it is easier to read tests in review. You can move it to mock files if you want.